components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/fee-mechanism.adoc

```diff
@@ -33,40 +33,73 @@ where:
 
 == When is the fee charged?
 
 The fee is charged atomically with the transaction execution on L2. The Starknet OS injects a transfer of the fee-related ERC-20, with an amount equal to the fee paid, the sender equal to the transaction submitter, and the sequencer as a receiver.
 
 [#fee_limitations]
-== Fee limitations
+== Transaction Fee limits
+
+[#v3_fee_limitations]
+=== v3 transactions
 
-Users can specify the maximum fee that they are willing to pay for a transaction.
+With v3 transactions, users specifiy the max amount and max price for each resource. At the time of writing, the only available resource is L1 gas. In the future, we will introduce L2 gas which will be used to price L2 work (as opposed to only charging for the proof verification in L1 gas, which is what happens today).
 
-include::partial$snippet_v3_max_price_sidebar.adoc[]
+[#deprecated_fee_limitations]
+=== Deprecated transactions (verion < 3)
 
-For transaction versions before v3 transactions, the `max_fee` field specifies the maximum fee.
+With older transactio versions, users specify the maximum fee that they are willing to pay for a transaction.
 
-The only limitation on the sequencer, which is enforced by the Starknet OS, is that the actual fee charged is bounded by `max_fee`, but for now, StarkWare's sequencer only charges the fee required to cover the proof cost, which is potentially less than the maximum fee.
+The only limitation on the sequencer, which is enforced by the Starknet OS, is that the actual fee charged is bounded by `max_fee`. While not enforced in the proof, the Starknet sequencer usually charges less than `max_fee`, as it charges exactly the amount given by the above fee formula.
 
-Currently, the sequencer only takes into account L1 costs involving proof submission. The following two components affect the L1 footprint of a transaction:
+[#what_do_we_pay_for]
+== What do we price
 
-* xref:#computation_without_builtins[Computational complexity]: A transaction's portion of the proof verification cost is proportional to its complexity.
-* xref:#onchain_data_components[Onchain data]: L1 calldata cost originating from xref:Network_Architecture/on-chain-data.adoc[data availability] and L2→L1 messages.
+At the time of writing, the following components are contributing to the transacation fee:
+
+* xref:#computation_without_builtins[Computational complexity]: The marginal cost of verifying the transaction on L1, measured in L1 gas.
+* xref:#onchain_data_components[Onchain data]: The cost of posting  the state diffs induced by the transaction to L1 (for more details, see xref:Network_Architecture/on-chain-data.adoc[data availability]). This is measured in L1 gas or L1 data gas, depending on whether or not the L2 block in which the transaction was included uses blobs or calldata.
+* L2→L1 messages: Messages sent to L1 are eventually sent to the Starknet core contract by the sequencer, which leads to the sending L2 transaction having additional L1 gas costs.
```

Comment:
I propose: "sent to the Starknet core contract as L1 calldata by the sequencer; therefore L2 transaction that send L2->L1 messages incur an additional L1 gas cost."